### PR TITLE
CB-11006 Fix the disk mounting logic to support Azure VM types withou…

### PR DESCRIPTION
**Pull request by daszabo**
…t resource/temporary disk

The mount and format scripts, in case of Azure, now dynamically determine the root, resource and managed disks.
Also reused the device names field of volumesets for Logical Unit Numbers, needed for the script to
properly pair devices to disks.

See detailed description in the commit message.

### CODE REVIEW

File: orchestrator-salt/src/main/resources/salt/salt/disks/mount/scripts/find-device-and-format.j2 [hunk 1]
```diff
@@ -104,6 +104,17 @@ save_env_vars_to_log_file() {
     log $LOG_FILE CLOUD_PLATFORM=$CLOUD_PLATFORM
 }
 
+get_temporary_disk_device_name() {
+  if [[ $CLOUD_PLATFORM -eq AZURE ]]; then
+    temporary_disk="/dev/$(ls -l /dev/disk/azure/ | awk '{print $9"\t"$11}' | grep -E '(resource[^-])' | cut -f2 | cut -f3 -d'/')"
+    log $LOG_FILE found temprary disk: \"$temporary_disk\"
```
Comment (lajosrodek) [inline orchestrator-salt/src/main/resources/salt/salt/disks/mount/scripts/find-device-and-format.j2:110]: Typo: temprary -> temporary.

File: orchestrator-salt/src/main/resources/salt/salt/disks/mount/scripts/find-device-and-format.j2 [hunk 2]
```diff
@@ -104,6 +104,17 @@ save_env_vars_to_log_file() {
     log $LOG_FILE CLOUD_PLATFORM=$CLOUD_PLATFORM
 }
 
+get_temporary_disk_device_name() {
+  if [[ $CLOUD_PLATFORM -eq AZURE ]]; then
+    temporary_disk="/dev/$(ls -l /dev/disk/azure/ | awk '{print $9"\t"$11}' | grep -E '(resource[^-])' | cut -f2 | cut -f3 -d'/')"
```
Comment (gergopapi2) [inline orchestrator-salt/src/main/resources/salt/salt/disks/mount/scripts/find-device-and-format.j2:109]: Question: what happens if there is no temporary disk? I think then the value of temporary_disk would be "/dev".
On AWS the return value is empty if there are no ephemeral disks. Will this difference cause no problems in the mounting? Will the script not want to mount the "/dev"?

**Review by gergopapi2 — COMMENTED** [review]
A general question at ephemerals, otherwise LGTM.

File: orchestrator-salt/src/main/resources/salt/bootstrapnodes/format-and-mount-common.sh [hunk 3]
```diff
@@ -83,6 +83,8 @@ get_root_disk() {
     root_partition=$(lsblk | grep /$ | cut -f1 -d' ' )
     if [[ $root_partition =~ "nvme" ]]; then
         echo "/dev/$(lsblk | grep /$ | cut -f1 -d' ' | sed 's/p\w//g' | cut -c 3-)"
+    elif [[ $CLOUD_PLATFORM -eq AZURE]]
```
Comment (pdarvasi) [inline orchestrator-salt/src/main/resources/salt/bootstrapnodes/format-and-mount-common.sh:86]: AZURE]] --> pls add extra whitespace

**Review by pdarvasi — APPROVED** [review]

**Comment by gergopapi2** [comment]
A comment that came up several time and can be addressed as a separate work item, I add it here not to forget: it would be great to have a test that would  test all instance types and just check if the scripts did work for them. This test could be potentially costly so it would be enough to run it once a week or so.

**Comment by biharitomi** [comment]
> A remark that came up several time and can be addressed as a separate work item, I add it here in order to not to forget: it would be great to have a test that would test all instance types and just check if the scripts did work for them. This test could be potentially costly so it would be enough to run it once a week or so.

I think you should create a JIRA item to cover it.

**Review by biharitomi — APPROVED** [review]
LGTM

**Review by biharitomi — APPROVED** [review]
LGTM

**Review by biharitomi — APPROVED** [review]
LGTM

**Review by sodre90 — APPROVED** [review]

**Pull request merged**